subprojects/resources-s3/src/main/java/org/gradle/internal/resource/transport/aws/s3/S3Client.java

```diff
@@ -42,6 +42,7 @@
 public class S3Client {
     private static final Logger LOGGER = LoggerFactory.getLogger(S3Client.class);
     private static final Pattern FILENAME_PATTERN = Pattern.compile("[^\\/]+\\.*$");
+    private static final Pattern DIRNAME_PATTERN = Pattern.compile("([^\\/]+)[\\/]$");
```

Comment:
Please remove unnecessary escaping and character groups from the regular expressions.